src/server/plugins/noopTracer.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2023 American Express Travel Related Services Company, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import fp from 'fastify-plugin';
+
+/**
+ * Fastify Plugin that injects a noop tracer into the request object
+ * @param {import('fastify').FastifyInstance} fastify Fastify instance
+ */
+const noopTracer = async (fastify) => {
+  const noopSpan = {
+    get addEvent() { return () => noopSpan; },
+    get end() { return () => {}; },
+    get isRecording() { return () => false; },
+    get recordException() { return () => {}; },
+    get setAttribute() { return () => noopSpan; },
+    get setAttributes() { return () => noopSpan; },
+    get setStatus() { return () => noopSpan; },
+    get spanContext() { return () => {}; },
+    get updateName() { return () => noopSpan; },
```

Comment:
why the use of `get` if `set` isn't use? I don't see the value of it

Reply:
That is the point. Using a getter only is a common pattern.